oscore-cf/src/main/java/org/eclipse/californium/oscore/OSCoreCtx.java

```diff
@@ -0,0 +1,673 @@
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Joakim Brorsson
+ *    Ludwig Seitz (RISE SICS)
+ *    Tobias Andersson (RISE SICS)
+ *    
+ ******************************************************************************/
+package org.eclipse.californium.oscore;
+
+import java.nio.ByteBuffer;
+import java.security.NoSuchAlgorithmException;
+import java.util.logging.Logger;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.eclipse.californium.core.coap.CoAP.Code;
+
+import com.upokecenter.cbor.CBORObject;
+
+import COSE.AlgorithmID;
+import COSE.CoseException;
+
+public class OSCoreCtx {
+
+	/**
+	 * The logger
+	 */
+	private static final Logger LOGGER = Logger.getLogger(OSCoreCtx.class.getName());
+
+	private AlgorithmID common_alg;
+	private byte[] common_master_secret;
+	private byte[] common_master_salt;
+	private byte[] common_iv;
+	private byte[] context_id;
+
+	private byte[] sender_id;
+	private byte[] sender_key;
+	private int sender_seq;
+
+	private byte[] recipient_id;
+	private byte[] recipient_key;
+	private int recipient_seq;
+	private int recipient_replay_window_size;
+	private int recipient_replay_window;
+
+	private AlgorithmID kdf;
+
+	private int rollback_recipient_seq = -1;
+	private int rollback_recipient_replay = -1;
+	private byte[] rollback_last_block_tag = null;
+
+	private byte[] last_block_tag = null;
+	private int seqMax = Integer.MAX_VALUE;
+
+	private int notification_number = -1;
+
+	private int id_length;
+	private int iv_length;
+	private int key_length;
+	
+	private Code CoAPCode = null;
+
+	/**
+	 * Constructor. Sets all context parameters. Mostly useful for testing.
+	 * 
+	 * @param masterSecret the master secret
+	 * @param alg the encryption algorithm as defined in COSE
+	 * @param senderId the sender id
+	 * @param senderKey the sender key
+	 * @param senderIV the sender IV
+	 * @param senderSeq the sender sequence number
+	 * @param recipientId the recipient id
+	 * @param recipientKey the recipient key
+	 * @param recipientIV the recipient IV
+	 * @param recipientSeq the recipient sequence number
+	 * @param replayWsize the reply window size
+	 * @param replayW the replay window start
+	 * @param kdf the key derivation function
+	 * @param masterSalt the optional master salt, can be null
+	 * 
+	 */
+	public OSCoreCtx(byte[] masterSecret, AlgorithmID alg, byte[] senderId, byte[] senderKey, int senderSeq,
+			byte[] recipientId, byte[] recipientKey, int recipientSeq, int replayWsize, int replayW, AlgorithmID kdf,
+			byte[] masterSalt, byte[] contextId) {
+
```

Comment:
This constructor requires mostly all parameters to be not null, and therefore differs from the other ones. It seems also to be not used. So, why is it required?

Reply:
It is used in interop testing, but those test cases are not included in this PR. I will remove this constructor for this PR and just put in when the interop tests are included.